Optional ERC721Consumable Extension (#4400)
ethereum/EIPs: EIPS/eip-4400.md

---
eip: 4400

> title: ERC721 Consumer Extension

Reviewer: ```suggestion
title: ERC-721 Consumer Extension
```

> description: Standard interface extension for ERC721 Consumer role  

Reviewer: ```suggestion
description: Standard interface extension for ERC-721 consumer role  
```

> author: Daniel Ivanov (@dani__ivanov) <daniel.k.ivanov95@gmail.com>

Reviewer: ```suggestion
author: Daniel Ivanov (@dani__ivanov)
```

discussions-to: https://ethereum-magicians.org/t/erc-4400-erc721consumer-extension/7371
status: Draft
type: Standards Track
category: ERC
created: 2021-10-30
requires: 721
---

## Abstract

This specification defines standard functions outlining a `consumer` role for instance(s)

> of [ERC-721](https://eips.ethereum.org/EIPS/eip-721). An implementation allows reading the current `consumer` for a

Reviewer: ```suggestion
of [ERC-721](./eip-721.md). An implementation allows reading the current `consumer` for a
```

Please link to all EIPs using this format.

given NFT (`tokenId`) along with a standardized event for when an `consumer` has changed. The proposal depends on and
extends the existing [ERC-721](https://eips.ethereum.org/EIPS/eip-721).

## Motivation

Many [ERC-721](https://eips.ethereum.org/EIPS/eip-721) contracts introduce their own custom role that grants permissions
for utilising/consuming a given NFT instance. The need for that role stems from the fact that other than owning the NFT

> instance, there are other actions that can be performed on an NFT. For example, various metaverses use`operator`

Reviewer: ```suggestion
instance, there are other actions that can be performed on an NFT. For example, various metaverses use `operator`
```

/`contributor`
roles for Land (ERC-721), so that owners of the land can authorise other addresses to deploy scenes to them (f.e.
commissioning a service company to develop a scene).

It is common for NFTs to have utility other than simply owning it. That being said, it requires a separate standardized
consumer role, allowing compatibility with user interfaces and contracts, managing those contracts.

> Having a `consumer` role will enable protocols to integrate and build on top of dApps that issue ERC721 tokens.

Reviewer: ```suggestion
Having a `consumer` role will enable protocols to integrate and build on top of dApps that issue ERC-721 tokens.
```

Reviewer: Please replace `ERC721` with `ERC-721` in all places.
